poglink/cogs/rates.py

```diff
@@ -66,14 +69,19 @@ def __init__(self, client):
     async def get_current_rates(url):
         logger.debug(f"Requesting current server rates")
         async with aiohttp.ClientSession() as session:
-            async with session.get(
-                url,
-                headers={"Pragma": "no-cache", "Cache-Control": "no-cache"},
-            ) as response:
-                response = await response.text()
-                rates = RatesStatus.from_raw(response)
-                logger.debug(f"Obtained server rates: {RatesStatus}")
-                return rates
+            try:
+                async with session.get(
+                    url,
+                    headers={"Pragma": "no-cache", "Cache-Control": "no-cache"},
+                ) as response:
+                    response = await response.text()
+                    rates = RatesStatus.from_raw(response)
+                    logger.debug(f"Obtained server rates: {RatesStatus}")
+                    return rates
+            except ValueError as e:
+                raise RatesProcessError(e) from e
+            except Exception as e:
+                raise RatesFetchError(e) from e
```

Comment:
I felt it was more appropriate to have these custom exceptions re-raised where they would first occur, rather than in an outer function like before. Plus that outer function (`compare_posted_rates()`) no longer exists